repr_test.go

```diff
@@ -57,6 +57,34 @@ func TestReprEmptySliceMapFields(t *testing.T) {
 	equal(t, `struct { S []string; M map[string]string; NZ []string }{NZ: []string{"a", "b"}}`, String(v, OmitEmpty(true)))
 }
 
+type zeroTest struct {
+	i int
+}
+
+var _ isZeroer = (*zeroTest)(nil)
+
+func (z zeroTest) IsZero() bool {
+	return z.i == 100
+}
+
+func TestReprZeroSliceMapFields(t *testing.T) {
+	v := struct {
+		ZSl []string
+		Sl  []string
+		ZM  map[string]string
+		M   map[string]string
+		ZI  int
+		I   int
+		ZS  string
+		S   string
+		ZB  bool
+		B   bool
+		ZC  zeroTest
+		C   zeroTest
+	}{[]string{}, []string{"a", "b"}, map[string]string{}, map[string]string{"a": "b"}, 0, 1, "", "a", false, true, zeroTest{i: 100}, zeroTest{i: 10}}
```

Comment:
For future reference, I think this would be easier to read with a concrete type declared in the function.

Reply:
Yeah probably :sweat_smile:.

Reply:
I think the only issue with that is I can't declare functions on those, can I?